src/linear_algebra/basic.lean

```diff
@@ -1922,13 +1973,12 @@ instance automorphism_group.to_linear_map_is_monoid_hom :
   map_mul := λ f g, rfl }
 
 /-- The group of invertible linear maps from `M` to itself -/
-def general_linear_group := units (M →ₗ[R] M)
+@[reducible] def general_linear_group := units (M →ₗ[R] M)
```

Comment:
Noob question: should definitions like this be `reducible` in general?

Reply:
Define "like this". This definition is slightly inconsistent with other parts of the library. I would use `linear_equiv M M` for this.

Reply:
Hmm, I guess I'm wondering which definitions of important mathematical objects should be reducible.

Reply:
I agree with Chris that this definition should probably be refactored. But even then, I think it would make sense to make this an `abbreviation` or `@[reducible] def`.

Reply:
I'd prefer to make this refactor in another PR to keep this one short.

Reply:
I agree. @ChrisHughes24 do you have other comments? Or shall we merge?

Reply:
Let's merge